### PR TITLE
Improve "upload" function

### PR DESCRIPTION
**Pull request by ViorelSuman**
Update "upload" function in order to be able to handle a specific
"upload" context (like upload a variable value, a byte array, etc.).

### CODE REVIEW

File: libuuu/fastboot.cpp [hunk 1]
```diff
@@ -266,8 +269,11 @@ int FBUpload::run(CmdCtx* ctx)
 	cmd.format("upload");
 
 	std::vector<uint8_t> buff;
-	if (fb.Transport(cmd, nullptr, buff.size(), &buff))
+	if (fb.Transport(cmd, nullptr, buff.size(), &buff)) {
+		string err = "Error on transport !";
+		set_last_err_string(err);
```
Comment (nxpfrankli) [inline libuuu/fastboot.cpp:277]: All error code setting at leaf node.  For here,  "error on transport" should be set in fb.transport() function, where actuall return -1.

**Review by nxpfrankli — CHANGES_REQUESTED** [review]
I have not seen uboot's upload implement, which supportted additional parameter?

If it is also added by you, do you have uboot's side patch?  Any defination about this additional parameter

**Comment by ViorelSuman** [comment]
> I have not seen uboot's upload implement, which supportted additional parameter?
> 
> If it is also added by you, do you have uboot's side patch? Any defination about this additional parameter

@nxpfrankli , I've added you as reviewer in the pull request I've made in our internal tree. On "uuu" side the optional "-v <some variable name>" just defines the non-default context in which the "upload" command is executed, the default behavior is still in place if "-v ..." option is not passed.

File: libuuu/fastboot.cpp (hunk 1 above)
Comment (ViorelSuman) [reply libuuu/fastboot.cpp]: Done.

**Pull request merged**